### PR TITLE
OSSM-3767 Refactor TestIngressWithoutTlsTermination (T10)

### PR DESCRIPTION
**Pull request by yxun**
This PR includes:
- Refactor  TestIngressWithoutTlsTermination (T10)
- Add an util method GetIngressGatewaySecurePort in pkg/util/istio/

### CODE REVIEW

**Review by luksa — APPROVED** [review]
LGTM. I just made a few small adjustments.

File: pkg/tests/tasks/traffic/ingress/ingress_without_tls_termination_test.go [hunk 1]
```diff
@@ -15,69 +15,58 @@
...
-		if strings.Contains(string(bodyByte), "Welcome to nginx") {
-			log.Log.Info(string(bodyByte))
-		} else {
-			t.Errorf("Failed to get Welcome to nginx: %v", string(bodyByte))
-		}
+		t.LogStep("Verify NGINX is reachable from outside the cluster through the ingressgateway")
+		gatewayHTTP := istio.GetIngressGatewayHost(t, meshNamespace)
+		secureIngressPort := istio.GetIngressGatewaySecurePort(t, meshNamespace)
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			curl.Request(t,
```
Comment (yxun) [inline pkg/tests/tasks/traffic/ingress/ingress_without_tls_termination_test.go:65]: @luksa  This curl request is failing in my OCP test.
I am getting curl failed to resolve error. Did the test pass from your side ?

> curl -v --resolve "nginx.example.com:$SECURE_INGRESS_PORT:$INGRESS_HOST" --cacert $CERT > "https://nginx.example.com:$SECURE_INGRESS_PORT"
> * Couldn't parse CURLOPT_RESOLVE entry 'nginx.example.com:443:istio-ingressgateway-istio-system.apps.yuaxu-maistra-daily.devcluster.openshift.com'!

Comment (luksa) [reply pkg/tests/tasks/traffic/ingress/ingress_without_tls_termination_test.go:65]: Yes, it works. But the error message you pasted here doesn't have sense. We're not even calling curl here, but making a request via Go's http package, so how can the error come from curl?

Comment (yxun) [reply pkg/tests/tasks/traffic/ingress/ingress_without_tls_termination_test.go:65]: I was troubleshooting it with a manual curl command.
Nvm, it must be my local network or DNS issue. I run it again and this PR test is Pass from my side now.
Thanks

**Pull request merged**